android/quest/src/androidTest/java/org/smartregister/fhircore/quest/integration/ui/main/components/TopScreenSectionTest.kt

```diff
@@ -48,7 +48,9 @@ class TopScreenSectionTest {
         title = "All Clients",
         searchText = "search text",
         onSearchTextChanged = mockListener,
-      ) {}
+        onTitleIconClick = {},
+        onFilterIconClick = {},
```

Comment:
Can we have one-click action with different event types?

Implement event wrapper

```kotlin 
sealed class ToolbarClickEvent {
   object FilterData : ToolbarClickEvent()
   object Navigate: ToolbarClickEvent()
 }
```

Update function parameter

```kotlin
fun TopScreenSection(
  modifier: Modifier = Modifier,
  title: String,
  searchText: String,
  searchPlaceholder: String? = null,
  toolBarHomeNavigation: ToolBarHomeNavigation = ToolBarHomeNavigation.OPEN_DRAWER,
  onSearchTextChanged: (String) -> Unit,
  onClick: (ToolbarClickEvent) -> Unit,
) {
``` 

Invoke event
```kotlin
  IconButton(onClick = { onTitleIconClick (ToolbarClickEvent.Navigate) } )
```

Finally update the logic to switch between the different click events.